version: "3.8"

services:

  postgres:
    image: postgres:latest
    volumes:
      - postgres:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password1234
      POSTGRES_DB: db
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $$POSTGRES_USER $$POSTGRES_DB"]
    ports:
      - "5432:5432"
    networks:
      - web_app

  django:
    build: .
    volumes:
      - ./:/app
    ports:
      - "8000:8000"
    environment:
      - DJANGO_DEBUG=1
    depends_on:
      - postgres
    networks:
      - web_app

  proxy:
    image: nginx:latest
    volumes:
      - ./default.conf.tpl:/etc/nginx/conf.d/default.conf
    ports:
      - "80:80"
    environment:
      - DOMAIN=${DOMAIN}
    depends_on:
      - django
    networks:
      - web_app

  # certbot:
  #   image: certbot/certbot:latest
  #   volumes:
  #     - /var/certbot/conf:/etc/letsencrypt/:rw
  #     - /var/certbot/www/:/var/www/certbot/:rw
  #   depends_on:
  #     - proxy
      
networks:
  web_app:

volumes:
  postgres:
